presidio-analyzer/presidio_analyzer/analyzer_engine_provider.py

```diff
@@ -54,42 +66,66 @@ def create_engine(self) -> AnalyzerEngine:
         supported_languages = self.configuration.get("supported_languages", ["en"])
         default_score_threshold = self.configuration.get("default_score_threshold", 0)
 
-        registry = self._load_recognizer_registry(supported_languages)
+        registry = self._load_recognizer_registry(
+            supported_languages=supported_languages, nlp_engine=nlp_engine
+        )
 
         analyzer = AnalyzerEngine(
             nlp_engine=nlp_engine,
             registry=registry,
             supported_languages=supported_languages,
-            default_score_threshold=default_score_threshold
+            default_score_threshold=default_score_threshold,
         )
 
-        analyzer.registry.add_nlp_recognizer(nlp_engine=analyzer.nlp_engine)
-
         return analyzer
 
-    def _load_recognizer_registry(self,
-                                  supported_languages: Optional[List[str]] = None
-                                  ) -> RecognizerRegistry:
-        if "recognizer_registry" not in self.configuration:
+    def _load_recognizer_registry(
+        self,
+        supported_languages: List[str],
+        nlp_engine: NlpEngine,
+    ) -> RecognizerRegistry:
+        if self.recognizer_registry_conf_file:
```

Comment:
Option 1: read from file
Option 2: read from the analyzer conf file (if it's there)
Option 3: default